### DISCUSSION TITLE
Cypress - Does cypress support real device automation test.

### DISCUSSION THREAD

**Discussion by jackchem2003** [cypress-io/cypress#27014] [Questions and Help]
Team,

We are using Cypress for Web automation and need to know whether Cypress support real device automation.
Specifically, on the Android and IOS devices. We need to perform web automation test on native apps.

We know that Ionic package support hybrid web automation. But, we are looking for solutions on native apps.
Please help us here.

Thanks,
Jagadeesh

**Comment by portyu9** [comment]
Hi @jackchem2003 

No. Cypress does **not** support automating native Android or iOS application UIs, including running tests against native apps on real mobile devices.

Cypress can test:

- normal web applications with mobile-sized viewports
- responsive websites
- applications whose UI actually runs in a browser/WebView, such as appropriate Ionic web applications

But it cannot drive native Android/iOS controls or provide real-device native-app automation.

For a genuinely native mobile application, you would need a mobile-specific automation framework such as Appium, Maestro, or a framework-specific solution such as Detox.

You can still use Cypress separately for the web portions/APIs of the same product, but not for the native device UI itself.

Please mark this answer as accepted if it helped, thank you.
